Jk  vy  share list

BikeMouse opened the pull request:
## Description

Implemented the "sharing list with another user by email" functionality

## Related Issue

closes #6 

## Acceptance Criteria

- [x]  The `ManageList` view shows a form that allows the user to enter an email to invite an _existing_ user to a list, in addition to the form that allows them to add items to that list.
- [x]  The input that accepts the email  has a semantic `label` element associated with it
- [x]  The user can submit this form with both the mouse and the `Enter` key
- [x]  If the other user exists, the user is alerted that the list was shared
- [x]  If the other user _does not_ exist, the user is shown an error message that explains the problem

## Type of Changes

adding feature

## Updates

### Before
![shareListBefore](https://github.com/the-collab-lab/tcl-71-smart-shopping-list/assets/99062030/23cce783-1a54-499c-b715-1229cb38966f)


### After
![successSharedList](https://github.com/the-collab-lab/tcl-71-smart-shopping-list/assets/99062030/65b74261-c65c-4bbd-a0d9-cb26cd521d90)
![errorSharedList](https://github.com/the-collab-lab/tcl-71-smart-shopping-list/assets/99062030/6b4549a9-19bf-41f8-907b-a5a16421c679)

## Testing Steps / QA Criteria

git checkout jk--vy--share-list
git pull
npm start
go to manage list
provide email

BikeMouse committed:
add form to managelist.jsx

vivitt committed:
change input type to email, add quotes to formData.get key

vivitt committed:
pass userId as prop to ManageList.jsx in App.jsx

vivitt committed:
pass userId as prop to ManageList.jsx in App.jsx

vivitt committed:
return response from shareList function in firebase.js

vivitt committed:
check if user email input has value before sendInvite

vivitt committed:
check if item input has value before adding a new item

vivitt committed:
clear mail input if it has no value

ocsiddisco reviewed (approved):
Nicely done! I like the checking for an empty user input!
Something that is not in the requirement, but could be a nice add: checking if the user is not assigning their own email address.

elitcenk commented:
It looks good. Well done.

BikeMouse commented:
@ocsiddisco Do you mean if I put my own email I get a message saying "You put in your own email?"

borjaMarti reviewed (approved):
Tested locally and works as expected, nice job!! 💪

vivitt committed:
- pass userEmail prop to check if user is entering their own email
- display an error message if the user tries to submit an empty string
- edit alert messages

vivitt commented:
I noticed that in our implementation, a user is only able to share a list if they are the owner of that list. I wonder if this is the expected behavior in the app, or if we might be missing some work to allow every user in the list to share it with others.

I created a new-test-list and shared it with all team members in case anyone wants to test it.

BikeMouse commented:
I could imagine that is the expected behaviour tbh. In fact, picturing myself using this app, I wouldn't want a random user to share my list

vivitt commented:
> I could imagine that is the expected behaviour tbh. In fact, picturing myself using this app, I wouldn't want a random user to share my list

Yes, I guess so... I was unsure about this since I tested the feature with `test-list` which it is also within my lists but was created by other user and wasn't able to share it.
